### PR TITLE
Fixed a desync involving WithCargo

### LINKED ISSUE #9094: Desync when Cargo.InitialUnits is used
Game Crash when i build modded Plane
 on this version: "Civillian Outrage BETA 48a"   http://resource.openra.net/maps/4168
out of sync 

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25838582-desync-when-cargo-initialunits-is-used?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Mailaender**
https://github.com/OpenRA/OpenRA/pull/9099 with performance fixes applied. Closes #9094.

### CODE REVIEW

**Comment by RoosterDragon** [comment]
:+1:

**Comment by chrisforbes** [comment]
:+1:

**Comment by chrisforbes** [comment]
@Mailaender I don't have time for stable branch wrangling today -- but I think you're good to go.

File: OpenRA.Mods.Cnc/Traits/Render/WithCargo.cs [hunk 1]
```diff
@@ -45,6 +45,19 @@ public WithCargo(Actor self, WithCargoInfo info)
...
 		}
 
+		public void Tick(Actor self)
+		{
+			if (facing == null)
+				return;
+
+			foreach (var c in cargo.Passengers)
+			{
+				var cargoFacing = c.TraitOrDefault<IFacing>();
```
Comment (pchote) [inline OpenRA.Mods.Cnc/Traits/Render/WithCargo.cs]: You can cache this when actors enter/leave the transport (compare with `AttackGarrisoned`) and avoid N trait lookups per tick.

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/2e57c45545103ec965a6a9dc107926a825effc1d...df5799eb2afe88036fab4089717854449b626c9a)

File: OpenRA.Mods.Cnc/Traits/Render/WithCargo.cs (hunk 1 above)
Comment (Mailaender) [inline OpenRA.Mods.Cnc/Traits/Render/WithCargo.cs]: Tried it in https://github.com/Mailaender/OpenRA/tree/with-cargo-perf but somehow the turrets vanish.

Comment (Mailaender) [inline OpenRA.Mods.Cnc/Traits/Render/WithCargo.cs]: Wait, that was actually unrelated. https://github.com/OpenRA/OpenRA/pull/9199

**Pull request merged**